tests/system_tests_distribution.py

```diff
@@ -3884,7 +3882,7 @@ def __init__(self,
         n_links_per_message = 4
         self.n_expected_transitions = len(self.senders) * self.messages_per_sender * n_links_per_message
 
-        self.debug = False
+        self.debug = debug
 
         self.test_name = test_name
 
```

Comment:
Since github won't let me comment on lines that haven't been changed: this is regards to the debug_print() function: add a flush=True to the print statement:  print(message, flush=True).   unittest has a tendency to not flush stdout if a failure is hit.